websockets/cext.c

```diff
@@ -0,0 +1,128 @@
+#define PY_SSIZE_T_CLEAN
+#include <Python.h>
+
+#include <stdlib.h> /* getenv */
+
+#ifndef HAS_SSE2
+/* setup HAS_SSE2 macro according to compiler predefined macros */
+#ifdef __SSE2__
+#define HAS_SSE2 1
+#else
+#define HAS_SSE2 0
+#endif
+#endif /* not defined HAS_SSE2 */
+
+#if HAS_SSE2
+#include <emmintrin.h>
+#endif
+
+#ifndef HAS_FAST_UNALIGNED_ACCESS
+/* This is not enabled by default */
+/* build with CFLAGS="-DHAS_FAST_UNALIGNED_ACCESS=1" to enable */
```

Comment:
> Given the code of cpython, the buffer will always be aligned properly.

I'm fairly comfortable relying on this property. I'm pretty sure common memory allocators make aligned allocations.

FWIW allocating a bytes object appears to require 32 bytes + the length of the string + 1 byte for the `\x00`:

```
>>> import sys
>>> sys.getsizeof(b'')
33
>>> sys.getsizeof(b'123')
36
```

I see little benefit in an unaligned allocation of this size.

----

Now I have to admit I'm lost:

- I didn't spend enough time reviewing your original commit yesterday to figure out what the branches with and without `HAS_FAST_UNALIGNED_ACCESS` did :-(
- that commit disappeared with the force push
- I'm not sure what assumption your latest iteration makes!

I'm sorry, my C is fairly weak, I'm doing my best but I need some hand-holding.

First I'm going to focus on adding the SSE optimization (which looks fun!) on top of #214 then we can revisit this as well.